frontend/src/Judge/components/VotePanel.js

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Alert, Button, ButtonGroup } from 'reactstrap'
+import styled from 'styled-components'
+
+const VoteSpacing = styled.span`
+  margin: 10px;
+`
+
+class VotePanel extends Component {
+  static propTypes = {
+    vote: PropTypes.func.isRequired,
+    data: PropTypes.shape({ 
+      vote: PropTypes.shape({
+        value: PropTypes.number
+      })
+    })
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      alertVisible: false
+    }
+  }
+
+  onDismiss() {
+    this.setState({ alertVisible: false })
+  }
+
+  handleVote() {
+    // TODO: Notify on errors
+    this.setState({ alertVisible: true })
+    setTimeout(() => {
+      this.setState({ alertVisible: false })
+    }, 3000);
+  }
+
+  render() {
+    const {
+      vote,
+      data
+    } = this.props
+    const entryVote = data.vote
+    return (
+      <div>
+        <ButtonGroup >
+          <VoteSpacing><Button color="primary" size="lg" active={entryVote && entryVote.value === 0} onClick={() => vote(0).then(this.handleVote())}>No</Button></VoteSpacing>
```

Comment:
Instead of the `<VoteSpacing>` block, could we just add 10px of margin to the `<Button>`? I _think_ this has the added advantage of an extended click-target... Bill managed to extend the click target area for the navigation buttons, but i'm not certain what method is appropriate here. I think it's OK to merge if the CSS gets too confusing though.